pgpubsub/listen.py

```diff
@@ -155,7 +164,6 @@ def _execute(self):
         channel = self.channel_cls.build_from_payload(
             self.notification.payload, self.callbacks)
         channel.execute_callbacks()
-        self.connection_wrapper.poll()
```

Comment:
Apparently this was the attempt to address this issue but the polling should be done after the last operation in the notification processing.